### PR TITLE
Commenting on a book

### PR DESCRIPTION
**Pull request by agoz17**
Commenting on a book is implemented here. The nodes and their transitions are embedded in our code as we decided. I also added new node transititions as dialog ending when Goodbye is written and also I added a specific situation that when there is no match from wit our bot says 'I couldn't understand can you express it more simple?'. I also made title is the only needed feature for a book to get the job done. The developer who implements the book api can easily change the relevant places in the code.

### CODE REVIEW

File: bookbotproject/conversationtree/views.py [hunk 1]
```diff
@@ -90,29 +93,61 @@ def add_new_user(request, _name, _userid, _chatid):
...
+        Book.objects.create(title=_title)
+        commentbook = Book.objects.get(title=_title)
+    _user = TelegramUser.objects.get(userid=_userid)
+    newComment = Comment.objects.create( user=_user,comment=_comment, book = commentbook)
+    return HttpResponse(status=200)
+
 def get_response(request, _message, _chatid):
     """ waits until a response from wit ai, it may take so much time !!!!"""
-
+    #print('wow')
```
Comment (theTytoAlba) [inline bookbotproject/conversationtree/views.py]: we can uncomment this with a more explanatory sentence or we can delete this line altogether

File: bookbotproject/conversationtree/views.py [hunk 2]
```diff
@@ -90,29 +93,61 @@ def add_new_user(request, _name, _userid, _chatid):
...
     if request.method == 'GET':
         intent_ret = None
-        while intent_ret == None :
+        intent_count = 0
+        while intent_ret == None and intent_count<6:
+            intent_count = intent_count + 1
+            #print(_message+'THISISINTENT')
```
Comment (theTytoAlba) [inline bookbotproject/conversationtree/views.py]: here too

File: bookbotproject/conversationtree/views.py [hunk 3]
```diff
@@ -90,29 +93,61 @@ def add_new_user(request, _name, _userid, _chatid):
...
-                curr_user.currentnode=curr_node.get_children()[i]
-                curr_user.save()
-                print("here")
-                return JsonResponse(curr_user.currentnode.message, safe=False)
-            
-    return JsonResponse(curr_user.currentnode.message, safe=False)
-
-    
+        #print(intent_ret)
+        #print(curr_node)
```
Comment (theTytoAlba) [inline bookbotproject/conversationtree/views.py]: more

File: bookbotproject/telegrambot.py [hunk 4]
```diff
@@ -80,24 +80,43 @@ def send_message(message, chat_id):
...
+			elif r.text == '\"Goodbye bookworm!\"':
+				send_message(r.text, chat);
+				counter = 0
+				end_switch = True
+			elif r.text == '\"Which book do you want to comment on?\"':
+				comment = text
+				send_message(r.text, chat);
+			elif r.text == '\"Your comment is saved!\"':
+				book = text
+				url =  HOST + "addComment/{}/{}/{}/".format(book, user_id, comment)
```
Comment (theTytoAlba) [inline bookbotproject/telegrambot.py:109]: neat. thank you.

**Review by theTytoAlba — COMMENTED** [review]
Can you delete the commented print statements? we should not merge to master with debug comments

File: bookbotproject/telegrambot.py (hunk 4 above)
Comment (agoz17) [reply bookbotproject/telegrambot.py:109]: I've deleted them Irmak. thank you for the review.

**Comment by melihmutlu** [comment]
I only checked telegrambot.py. It looks great. Thanks @agoz17

**Comment by theTytoAlba** [comment]
This looks good to me, merging.

**Pull request merged**